eng/pipelines/templates/stages/archetype-java-release.yml

```
@@ -116,7 +116,38 @@ stages:
                     PackageName: ${{artifact.name}}
                     ServiceName: ${{parameters.ServiceDirectory}}
                     ForRelease: true
-
+          - job: InstallationCheck
+            displayName: "Installation Check"
+            variables:
+              - template: ../variables/globals.yml
+            steps:
+              - download: current
+                artifact: ${{parameters.ArtifactName}}-signed
+              - pwsh: |
+                  Copy-Item "$(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed/${{artifact.groupId}}/${{artifact.name}}/" "$(Pipeline.Workspace)/install-check-dir/${{artifact.name}}" -Recurse
+                  Write-Host "Copying artifact ${{artifact.name}}"
+                  $pomf = Get-ChildItem $(Pipeline.Workspace)/install-check-dir/${{artifact.name}}/*.pom
+                  Rename-Item "$($pomf)" pom.xml
+                  Write-Host "Renaming $($pomf) to pom.xml"
```

It would be better to log the command actions before actually running them, otherwise they might fail and exit the script first.

```suggestion
                  Write-Host "Copying artifact ${{artifact.name}}"
                  Copy-Item "$(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed/${{artifact.groupId}}/${{artifact.name}}/" "$(Pipeline.Workspace)/install-check-dir/${{artifact.name}}" -Recurse
                  Rename-Item "$($pomf)" pom.xml
                  $pomf = Get-ChildItem $(Pipeline.Workspace)/install-check-dir/${{artifact.name}}/*.pom
                  Write-Host "Renaming $($pomf) to pom.xml"
```

---

To be more thorough, we generally prefer to log the actual command being run, which makes debugging easier and also helps to inspect paths that may be generated improperly.